### PR TITLE
Vungle 6.4.3 MoPub Adapter Implementations

### PR DESCRIPTION
**Pull request by YueVungle**
In Vungle 6.4.3 MoPub Adapter, we have implemented below features: 
1、Add MREC banner ad type.
2、Support Vungle 6.4.3 SDK
3、Support MoPub 5.8.0 SDK
4、Add trackClick event for banner ads

### CODE REVIEW

File: Vungle/VungleBannerCustomEvent.m [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+- (void)vungleAdDidAppear {
+    MPLogInfo(@"Vungle video banner did appear");
+}
+
+- (void)vungleAdWillDisappear
+{
+    MPLogInfo(@"Vungle video banner will disappear");
+}
+
+- (void)vungleAdDidDisappear
```
[the remark this answers is not in the document]
Comment (dravyan) [reply Vungle/VungleBannerCustomEvent.m]: @ashinagawa , yes, we can remove them from here. Thanks!

[the remark this answers is not in the document]
Comment (YueVungle) [reply Vungle/VungleBannerCustomEvent.m]: @dravyan We have removed vungleAdWillAppear, vungleAdDidAppear, vungleAdWillDisappear, vungleAdDidDisappear mehods. :)

File: Vungle/VungleBannerCustomEvent.m [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        loadFailError = error;
+        MPLogInfo(@"Vungle video banner failed to load with error: %@", error.localizedDescription);
+    }
+
+    [self.delegate bannerCustomEvent:self didFailToLoadAdWithError:loadFailError];
+}
+
+- (void)vungleAdDidFailToPlay:(NSError *)error
+{
+    // verify - is this necessary?
```
[the remark this answers is not in the document]
Comment (YueVungle) [reply Vungle/VungleBannerCustomEvent.m]: @dravyan We have removed vungleAdDidFailToPlay: method, please review the change. :)

**Comment by YueVungle** [comment]
@dravyan Thanks for your comments and we have removed the redundant methods, the left we keep them is useful for us. Please review these updates.  O(∩_∩)O

File: Vungle/VungleBannerCustomEvent.m [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        }
+    }];
+    
+    [[VungleRouter sharedRouter] requestBannerAdWithCustomEventInfo:info size:size delegate:self];
+}
+
+- (void)dealloc {
+    // call router event to transmit close to SDK for report ad finalization / clean up
+//    [[VungleRouter sharedRouter] completeBannerAdViewForPlacementID:self.placementId];
+}
```
Comment (dravyan) [inline Vungle/VungleBannerCustomEvent.m]: Should be cleaning up delegates here?

**Comment by YueVungle** [comment]
@dravyan Hi, dravyan, for the cleaning up delegates, we will add "invalidate" method in 6.4.5.0 branch and do cleaning up work in this method. Thanks. 🙂

**Comment by dravyan** [comment]
@ashinagawa, @YueVungle,  closing this PR in favor of https://github.com/mopub/mopub-ios-mediation/pull/222

**Pull request closed without merging**